src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -194,6 +211,107 @@ def restrict(self: _BaseNdArrayField, domain) -> _BaseNdArrayField:
 
     __pow__ = _make_binary_array_field_intrinsic_func("pow", "power")
 
+    @overload
+    def __getitem__(self, index: DomainSlice) -> common.Field:
+        """Absolute slicing with dimension names."""
+        ...
+
+    @overload
+    def __getitem__(self, index: tuple[slice | int, ...]) -> common.Field:
+        """Relative slicing with ordered dimension access."""
+        ...
+
+    @overload
+    def __getitem__(
+        self, index: Sequence[common.NamedIndex]
+    ) -> common.Field | core_defs.DType[core_defs.ScalarT]:
+        # Value in case len(i) == len(self.domain)
+        ...
```

Comment:
this overload was only to emphasize the construction of lower dimensions, but not correct for mypy because the parameter is a subset of the first overload.
```suggestion
```